sql/parser/normalize.go

```diff
@@ -97,8 +97,11 @@ func (expr *ComparisonExpr) normalize(v *normalizeVisitor) Expr {
 				// The left side is const and the right side is a binary expression or a
 				// variable. Flip the comparison op so that the right side is const and
 				// the left side is a binary expression or variable.
-				expr.Operator = invertComparisonOp(expr.Operator)
-				expr.Left, expr.Right = expr.Right, expr.Left
+				*expr = ComparisonExpr{
+					Operator: invertComparisonOp(expr.Operator),
+					Left:     expr.Right,
+					Right:    expr.Left,
+				}
```

Comment:
it's not equivalent - the function cache is cleared here, but was not previously